feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -578,4 +600,40 @@ class BillingWrapper(
         Throwable().printStackTrace(printWriter)
         return stringWriter.toString()
     }
+
+    private fun BillingClient.querySkuDetailsAsyncEnsuringOneResponse(
+        params: SkuDetailsParams,
+        listener: SkuDetailsResponseListener
+    ) {
+        var hasResponded = false
+        querySkuDetailsAsync(params) { billingResult, skuDetailsList ->
+            if (hasResponded) {
+                log(
+                    LogIntent.GOOGLE_ERROR,
+                    OfferingStrings.EXTRA_QUERY_SKU_DETAILS_RESPONSE.format(billingResult.responseCode)
+                )
+                return@querySkuDetailsAsync
+            }
+            listener.onSkuDetailsResponse(billingResult, skuDetailsList)
+            hasResponded = true
```

Comment:
yes, I think that makes sense. This will happen in a thread, and whatever is executed in `listener.onSkuDetailsResponse` would be in that thread. In the meantime, another thread can call this lambda function and will see hasResponded as false. Maybe it all happens in the same thread but I don't think we can't guarantee it.

I think the reason we did it at the end is so that we make sure everything has been executed in the listener and in case there could be an exception in the listener, hasResponded is still `false`. But I think in the case of the exception, the developer should be the one retrying... I don't know if that makes sense.

This makes me think though... Should we synchronize access to `hasResponded`?